cli/src/action/database/upgrade.rs

```diff
@@ -0,0 +1,492 @@
+                    }
+                })
+                .collect::<Result<Vec<_>, _>>()?;
+
+            to.add_circuit(circuit, members)
+                .map_err(|e| ImportError::Circuit(id, e))?;
+            import_result.nodes += 1;
+        }
+
+        let proposals = from
+            .list_proposals(predicates)
+            .map_err(ImportError::Store)?;
+        for proposal in proposals {
+            let requester = proposal.requester_node_id().to_string();
+            to.add_proposal(proposal)
+                .map_err(|e| ImportError::Proposal(requester, e))?;
+            import_result.proposals += 1;
+        }
+
+        Ok(import_result)
+    }
+
+    /// Import yaml state from the specified directory to a database
+    fn import_yaml_state_to_database<P: Into<PathBuf>>(
+        state_dir: P,
+        db_store: &'_ dyn AdminServiceStore,
+    ) -> Result<(), CliError> {
+        info!("Loading YAML datastore... ");
+        fn invalid_utf8() -> CliError {
+            CliError::ActionError("'state_dir' is not a valid UTF-8 string".to_string())
+        }
+
+        let state_dir: PathBuf = state_dir.into();
+        let circuits_location = state_dir.join(CIRCUITS_FILE);
+        let proposals_location = state_dir.join(PROPOSALS_FILE);
+
+        if !(circuits_location.exists() || proposals_location.exists()) {
+            return Err(CliError::ActionError(
+                "no yaml state files found to import".to_string(),
+            ));
+        }
+
+        let yaml_admin_service_store = YamlAdminServiceStore::new(
+            circuits_location
+                .to_str()
+                .ok_or_else(invalid_utf8)?
+                .to_string(),
+            proposals_location
+                .to_str()
+                .ok_or_else(invalid_utf8)?
+                .to_string(),
+        )
+        .map_err(|err| {
+            CliError::ActionError(format!("unable to create YamlAdminServiceStore: {}", err))
+        })?;
+
+        info!("Processing import data... ");
+        let result = Self::import_store(db_store, &yaml_admin_service_store).map_err(|e| {
+            CliError::ActionError(match e.source() {
+                Some(source) => format!("{}: {}", e.to_string(), source),
+                None => e.to_string(),
+            })
+        })?;
+
+        info!("Backing up state files... ");
+        let new_circuits_location = circuits_location.with_extension("yaml.old");
+        let new_proposals_location = proposals_location.with_extension("yaml.old");
+        fs::rename(circuits_location, new_circuits_location)
+            .map_err(|e| CliError::ActionError(format!("could not move circuits file: {}", e)))?;
+        fs::rename(proposals_location, new_proposals_location)
+            .map_err(|e| CliError::ActionError(format!("could not move proposals file: {}", e)))?;
+
+        info!(
+            "Successfully imported {} node(s) and {} proposal(s)",
```

Comment:
Should nodes be circuits (in the log and in the import results)?

Reply:
Yep! Fixed